src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -1172,6 +1199,31 @@ public Integer call() throws Exception {
         tileWidth, tileHeight);
     }
 
+    // if the tmpdir is mounted as noexec, then any native library
+    // loading (OpenCV, turbojpeg, etc.) is expected to fail, which
+    // can cause conversion to fail with an exception that is not user friendly
+    //
+    // try to detect this case early and fail before the conversion begins,
+    // with a more informative message
+
+    // a temp file is created and set as executable
+    // in the noexec case, setting as executable is expected to silently fail
+    File tmpdirCheck = File.createTempFile("noexec-test", ".txt");
+    // expect 'success' to be true in the noexec case, even though
+    // the file will not actually be executable
+    boolean success = tmpdirCheck.setExecutable(true);
+    if (!success || !tmpdirCheck.canExecute()) {
+      String msg = System.getProperty("java.io.tmpdir") +
+        " is noexec; fix it or specify a different java.io.tmpdir";
+      if (getWarnNoExec()) {
+        LOGGER.warn(msg);
+      }
+      else {
+        throw new RuntimeException(msg);
+      }
+    }
+    tmpdirCheck.delete();
```

Comment:
In the scenario where `/tmp` is mounted as `noxec` and `--warn-no-exec` is not passed to the command, this deletion will never be executed as per the `throw` call a few lines earlier. Would it make sense to wrap this in a `try/finally` block to ensure temporary test files are always cleaned up?